client/layout/guided-tours/button-labels.jsx

```diff
@@ -0,0 +1,40 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import { identity } from 'lodash';
+
+// Returns React component with a localized label and optional icon
+function button( label, icon ) {
+	return localize( ( { translate: tr } ) => (
+		<strong>
+			{ icon }
+			{ icon && '\u00A0' /* NBSP between icon and label */ }
+			{ tr( label ) }
+		</strong>
+	) );
+}
+
+// Localized texts need to be wrapped in a `translate` function to be found by the l10n bot
+const translate = identity;
```

Comment:
can you explain why we use identity instead of the translate from `i18n-calypso`?

Reply:
Oh I think I get it. Is there some code that statically scans the codebase for `translate( string )` calls then generates translation files?

This code is sort-of tricking it because it isn't the real translate function -- you've swapped it with a more performant identity function.  Is that whats going on here? Because we shouldn't need to hack it that way -- ideally we could make translate much more performant for the en-->en case.

Reply:
We don't want to use the `translate` function imported from `i18-calypso`, but rather the prop supplied by the `localize` HoC:
```jsx
import { localize } from 'i18n-calypso';
const LocalizedComponent = localize( props => (
  <div>
    { props.translate( 'Settings' ) }
  </div>
) );
```

The main reason is that when the locale changes, the `localize` HoC listens to the locale change event and will rerender the wrapped component. The `translate` function return value will be different, and therefore the render output will be different, too.

> Is there some code that statically scans the codebase for instances of translate( string ) then generates translation files?

Yes, that's exactly what happens. To make it work at runtime, just passing a string to the `buttonLabel` function would be sufficient:
```js
buttonLabel( 'Settings' );
```

But the l10n tools scan for instances of `translate( string )`. The `string` cannot be a variable (the tool cannot interpret JS and figure out the variable value), but it must be a string literal. The syntax therefore needs to look like this:
```js
buttonLabel( translate( 'Settings' ) );
```

And then it follows that `translate` should behave like identity function here...